packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -296,20 +305,20 @@ export class IndexedDbPersistence implements Persistence {
       });
   }
 
-  shutdown(deleteData?: boolean): Promise<void> {
+  async shutdown(deleteData?: boolean): Promise<void> {
     if (!this.started) {
       return Promise.resolve();
     }
     this.started = false;
     this.clientMetadataRefresher.cancel();
     this.detachVisibilityHandler();
     this.detachWindowUnloadHook();
-    return this.releasePrimaryLeaseIfHeld().then(() => {
-      this.simpleDb.close();
-      if (deleteData) {
-        return SimpleDb.delete(this.dbName);
-      }
-    });
+    await this.releasePrimaryLeaseIfHeld();
+    await this.removeClientMetadata();
```

Comment:
FWIW it's probably unlikely that this code will actually run to completion during tab close...  So we shouldn't rely on it.  And if we want to increase the probability of it running, we may want to use Promise.all() or something so things can at least go in parallel (uh, maybe? I guess they both will acquire IndexedDb locks... so really we'd need to refactor into the same transaction).

But if we really want failover to be fast then we need to only rely on localStorage like we do with the zombie thing...  Feel free to just add a TODO(multitab) if you want to do this but not right now.